tests/Persistence/ElasticQueryRunnerTest.php

```diff
@@ -0,0 +1,75 @@
+<?php
+
+declare( strict_types = 1 );
+
+namespace ProfessionalWiki\WikibaseFacetedSearch\Tests\Persistence;
+
+use CirrusSearch\Connection;
+use CirrusSearch\SearchConfig;
+use Elastica\Client;
+use Elastica\Response;
+use MediaWiki\MediaWikiServices;
+use MediaWikiIntegrationTestCase;
+
+/**
+ * @group Database
+ */
+class ElasticQueryRunnerTest extends MediaWikiIntegrationTestCase {
+
+	public function testCanQueryElastic(): void {
+		$query = [
+			'query' => [
+				'match_all' => new \stdClass()
+			]
+		];
+
+		$resultSet = $this->runQuery( $query )->getData();
+
+		$this->assertArrayHasKey( 'hits', $resultSet );
+		$this->assertGreaterThan( 0, $resultSet['hits']['total']['value'] );
+	}
+
+	private function runQuery( array $query ): Response {
+		return $this->getClient()->request( '_search', 'GET', $query );
+	}
+
+	private function getClient(): Client {
+		return $this->getConnection()->getClient();
+	}
+
+	private function getConnection(): Connection {
+		return new Connection( $this->getSearchConfig() );
+	}
+
+	private function getSearchConfig(): SearchConfig {
+		/**
+		 * @var SearchConfig $config
+		 */
+		$config = MediaWikiServices::getInstance()->getConfigFactory()->makeConfig( 'CirrusSearch' );
+		return $config;
+	}
+
+	public function testCanQueryMediaWikiPage(): void {
+		$query = [
+			'query' => [
+				'bool' => [
+					'must' => [
+						[
+							'match' => [
+								'title' => 'Main Page'
+							]
+						]
+					]
+				]
+			]
+		];
+
+		$resultSet = $this->runQuery( $query )->getData();
+
+		$this->assertSame(
+			"MediaWiki has been installed. Consult the User's Guide for information on using the wiki software.",
```

Comment:
I tried to make this test more environment state independent but didn't get it to work in 30 minutes, so am moving on to higher priority tasks. WIP attempt at https://github.com/ProfessionalWiki/WikibaseFacetedSearch/commit/ff221c10b13322345908772f8d0dcc4ae7cae395